aws/terminator/application_services.py

```diff
@@ -367,3 +367,41 @@ def name(self):
 
     def terminate(self):
         self.client.delete_document(Name=self.name)
+
+
+class CloudFrontDistribution(Terminator):
```

Comment:
When there are no distributions to be listed the API returns an empty dictionary instead of an empty `StreamingDistributionList` or empty list of `Items`.  Meaning,
 
`client.get_paginator('list_streaming_distributions').paginate().build_full_result()['StreamingDistributionList']['Items']` returns a KeyError if there are no distributions (same for streaming distributions).  

For example, 
```
response = client.get_paginator('list_streaming_distributions').paginate().build_full_result() 
print(response)
{}
```
We'll need to execute the paginator, and then check to see if it contains any keys before deciding what to return in the create method of each terminator class.